### DISCUSSION TITLE
Suggestion of Secret service using etcd

### DISCUSSION THREAD

**Discussion by jihyungSong** [cloudforet-io/community#17] [Ideas]
**Propose Date** : 2022-11-17
**Proposer** : @jihyungSong  @whdalsrnt 

---

Currently, Cloudforet is to store sensitive data such as the user's Access Key in a more safety way through the Secret micro service.
(https://github.com/cloudforet-io/secret) 

About the Secret service, it is supported using various backend systems.

* [AWS Secrets Manager](https://aws.amazon.com/ko/secrets-manager/)
* [Consul by Hashicorp](https://www.consul.io/)
* [Vault by Hashicorp](https://www.vaultproject.io/)

I suggest to use **etcd** as a backend for Secret service.
(https://etcd.io/)

**etcd** is a strongly consistent, distributed key-value store that provides a reliable way to store data that needs to be accessed by a distributed system or cluster of machines.

Since most of the data by the users is in the key-values type, it is very suitable to use **etcd**, and Kubernetes also uses **etcd** by default, it is good to integrate with Cloudforet.

Basically, **etcd** is not stored in an encrypted, but it is not difficult to implement encryption and decryption of data before being stored in **etcd** through the handler structure of the Secret service.

**Comment by jihyungSong** [comment]
An issue related to etcd connector development has been generated.
* https://github.com/cloudforet-io/secret/issues/12
